modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/validators.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export function validateAny() {
+	return true;
+}
+
+export function validateArray(value, name, context) {
+	return validateType('array', value, name, context);
+}
+
+export function validateString(value, name, context) {
+	return validateType('string', value, name, context);
+}
+
+export function validateObject(value, name, context) {
+	return validateType('object', value, name, context);
```

Comment:
I don't know what the old implementation did, but are we cool with the new implementation considering `null` to be a valid object? (seeing as `typeof null` is `'object'`).

Reply:
Yea that's an oversight right there, I'll check out the details tomorrow, but at a quick glance this looks to be about as deep as it goes in the old implementation [validators.js#L248-L250](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-state/src/validators.js#L248-L250). Perhaps it has some fallbacks in `State.js` but don't know right now. I'll test and confirm tomorrow.

Reply:
Because it [goes through `maybe()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-state/src/validators.js#L270-L272) a quick reading suggests that `null` will fall through to the hardcoded `true` case and pass validation. (`undefined` will too, unless I'm misreading it.)

Reply:
Handled in https://github.com/liferay-frontend/liferay-portal/pull/853/commits/f2bb06ea364f2b59625a0eb326ce1b354d5c6a18